main/src/com/google/refine/operations/column/ColumnAdditionByFetchingURLsOperation.java

```diff
@@ -213,66 +217,89 @@ public void run() {
                 _project.processManager.onFailedProcess(this, new Exception("Another column already named " + _newColumnName));
                 return;
             }
-            
+
             List<CellAtRow> urls = new ArrayList<CellAtRow>(_project.rows.size());
-            
+
             FilteredRows filteredRows = _engine.getAllFilteredRows();
             filteredRows.accept(_project, createRowVisitor(urls));
-            
+
             List<CellAtRow> responseBodies = new ArrayList<CellAtRow>(urls.size());
             for (int i = 0; i < urls.size(); i++) {
                 CellAtRow urlData = urls.get(i);
-                long start = System.currentTimeMillis();
-                CellAtRow cellAtRow = fetch(urlData);
+                CellAtRow cellAtRow = cachedFetch(urlData);
                 if (cellAtRow != null) {
                     responseBodies.add(cellAtRow);
                 }
-                
+
                 _progress = i * 100 / urls.size();
-                try {
-                    long delay = _delay - (System.currentTimeMillis() - start);
-                    if (delay > 0) {
-                        Thread.sleep(delay);
-                    }
-                } catch (InterruptedException e) {
-                    if (_canceled) {
-                        break;
-                    }
-                }
+
+                if (_canceled) {
+                    break;
+	        }
             }
-            
+
+	    _urlCache.clear();
```

Comment:
Why would it help to remove entries when the key is no longer in use? Here the keys are strings or urls that are constructed just before the request, and are no longer referenced anywhere after we continue to the next row… So if the GC is a bit aggressive, this could effectively keep the cache empty, I think.
I am not an expert but using WeakHashMap here sounds a lot like the misunderstanding described here: https://stackoverflow.com/questions/1802809/javas-weakhashmap-and-caching-why-is-it-referencing-the-keys-not-the-values#1802879